projects/virtual-machine-service-typescript/src/hard-coded-contracts/processor.ts

```diff
@@ -0,0 +1,70 @@
+import { types } from "orbs-common-library";
+import StateCache from "../state-cache";
+import {
+    BaseContractStateAccessor,
+    ContractStateReadOnlyAccessor,
+    ContractStateReadWriteAccessor
+} from "../contract-state-accessor";
+
+import BaseSmartContract from "./base-smart-contact";
+
+interface CallRequest {
+    sender: string;
+    argumentsJson: string;
+    contractAddress: string;
+    lastBlockId?: number;
+}
+
+export default class HardCodedSmartContractProcessor {
+    contractAddresses = new Map<string, any>();
+    stateStorageClient: types.StateStorageClient;
+
+    constructor(stateStorageClient: types.StateStorageClient) {
+        this.stateStorageClient = stateStorageClient;
+
+        this.registerContract("bar-smart-contract", "barbar");
```

Comment:
This doesn't feel outside of the testing suite, but I'm sure we will refactor it soon enough.

Reply:
@lbeder you mean doesn't feel right?

Reply:
Yeah...

Reply:
We'll do. It's just that it's not currently possible to register contracts externally to the virtual machine service (they are hard-coded and pre-loaded).